### PR TITLE
Check for missing open source template files

### PR DESCRIPTION
**Pull request by cfarm**
Use node to check for missing open source template files and add them to your project as needed. Also activates .env file if you have one, adds it if you don't
## Additions
- `check-it.js` checks for missing open source files including TERMS, CHANGELOG, CONTRIBUTING
- also looks for `.env` file in prep for pulling in the [license-exceptions](https://github.com/cfarm/license-exceptions) module in future PR
- package.json adds dependencies for the above to function
- options/flags for running on command line: `--fix` to automagically add missing files; `--scrub` to run the scrub function (tba, just checks for `.env` file right now)
- thank you [vax](https://github.com/cfpb/vax/) for basically doing exactly what i needed this to do already 👏 
## Testing
- `npm install`
- run `npm link` so the command will work
- do one of these:

`check-it --fix --scrub .`

or try it without the optional flags

`check-it .`

or try it on another directory

`check-it --fix --scrub ../path/to/another/repo/`
## Review
- @contolini 

[Preview this PR without the whitespace changes](?w=0)
## Todos
- update readme so someone can actually use this
## Checklist
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
- [ ] Passes all existing automated tests
- [ ] New functions include new tests
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged
- [ ] Visually tested in supported browsers and devices
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

File: cmds/check-it.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    case 4:
+      if (fix) {
+        request('https://raw.githubusercontent.com/cfpb/open-source-checklist/master/.env_SAMPLE')
+          .on('error', function(err) {
+            cb(err, null)
+          })
+          .pipe(fs.createWriteStream(path.join(location, './.env')))
+        return cb(null, 'No .env file found. I created one for you. Now add it to your project\'s .gitignore! Then you can update the GHE_URL variable.');
+      }
+      cb('No .env file found. Please create one based on this sample, add it to your project\'s .gitignore,and update the GHE_URL variable: https://github.com/cfpb/open-source-checklist/blob/master/.env_SAMPLE', null);
```
Comment (contolini) [inline cmds/check-it.js]: Missing a space: `.gitignore,and update`

File: cmds/check-it.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    cb(null, 'Your CHANGELOG.md file is good to go.');
+  });
+}
+
+function scrubGHEReferences(cb) {
+  if (scrub) {
+    fs.readFile(path.join(location, './.env'), 'utf8', function (err, data) {
+      if (err) return handleIssue(4, cb);
+      var envPath = path.join(location, './.env');
+      shell.exec('. ' + envPath);
```
Comment (contolini) [inline cmds/check-it.js:65]: What's the purpose of executing the `.env` file?

File: cmds/check-it.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+      }
+      cb('No CHANGELOG.md file found. Please create one based on the Open Source Project Template: https://github.com/cfpb/open-source-project-template/blob/master/CHANGELOG.md', null);
+      break;
+    case 4:
+      if (fix) {
+        request('https://raw.githubusercontent.com/cfpb/open-source-checklist/master/.env_SAMPLE')
+          .on('error', function(err) {
+            cb(err, null)
+          })
+          .pipe(fs.createWriteStream(path.join(location, './.env')))
```
Comment (contolini) [inline cmds/check-it.js]: You can add `.env` to their gitignore for them with:

```
fs.appendFileSync(path.join(location, '.gitignore'), '.env', 'utf8');
```

**Comment by contolini** [comment]
👍 Very cool tool!

File: cmds/check-it.js (hunk 2 above)
Comment (cfarm) [reply cmds/check-it.js:65]: to set the GHE_URL variable and run the scrub git history searches w/it. let me know if you know a better way

Comment (cfarm) [reply cmds/check-it.js:65]: also: scrub function doesn't actually do shit rn except execute the `.env`

**Pull request merged**